pycardano/utils.py

```diff
@@ -235,35 +235,54 @@ def min_lovelace_post_alonzo(output: TransactionOutput, context: ChainContext) -
 
 
 def script_data_hash(
-    redeemers: Redeemers,
-    datums: List[Datum],
+    redeemers: Optional[Redeemers] = None,
+    datums: Optional[Union[List[Datum], NonEmptyOrderedSet[Datum]]] = None,
     cost_models: Optional[Union[CostModels, Dict]] = None,
 ) -> ScriptDataHash:
     """Calculate plutus script data hash
 
     Args:
-        redeemers (Redeemers): Redeemers to include.
-        datums (List[Datum]): Datums to include.
+        redeemers (Optional[Redeemers]): Redeemers to include.
+        datums (Optional[Union[List[Datum], NonEmptyOrderedSet[Datum]]]): Datums to include.
         cost_models (Optional[CostModels]): Cost models.
 
     Returns:
         ScriptDataHash: Plutus script data hash
     """
+    # Handle empty redeemers case - should be encoded as an empty map (A0 in hex)
     if not redeemers:
+        redeemer_bytes = cbor2.dumps({}, default=default_encoder)
```

Comment:
We should just check if the redeemer is actually None, and if so, assign an empty redeemer instance to it: `redeemers = Redeemers()`, similar to the implementation [here](https://github.com/Python-Cardano/pycardano/blob/30712f48e32d73fd5fe9e4eec5e70131d1623510/pycardano/txbuilder.py#L585.), and the encoding will be automatically handled.

```suggestion
        if redeemers is None:
            redeemers = Redeemers()
```

Empty `Redeemers()` will be encoded as `A0`.